bot/seasons/easter/egg_hunt/cog.py

```diff
@@ -0,0 +1,511 @@
+
+        if user.bot:
+            return False
+        if reaction.message.id != self.message.id:
+            return False
+        if reaction.emoji != self.egg:
+            return False
+        return True
+
+    async def collect_reacts(self, reaction: discord.Reaction, user: discord.Member):
+        """Handles emitted reaction_add events via listener."""
+
+        if not self.is_valid_react(reaction, user):
+            return
+
+        team = get_team_role(user)
+        if not team:
+            log.debug(f"Assigning a team for {user}.")
+            user = await assign_team(user)
+
+        if not self.first:
+            log.debug(f"{user} was first to react to egg on {self.message.id}.")
+            self.first = user
+            await self.start_timeout()
+        else:
+            self.users.append(user)
+
+    async def start(self):
+        """Starts the egg drop session."""
+
+        log.debug(f"EggHunt session started for message {self.message.id}.")
+        bot.add_listener(self.collect_reacts, name="on_reaction_add")
+        with contextlib.suppress(discord.Forbidden):
+            await self.message.add_reaction(self.egg)
+        self.timeout_task = asyncio.create_task(self.start_timeout(300))
+
+
+class SuperEggMessage(EggMessage):
+    """Handles a super egg session."""
+
+    def __init__(self, message: discord.Message, egg: discord.Emoji, window: int):
+        super().__init__(message, egg)
+        self.window = window
+
+    async def finalise_score(self):
+        """Sums and actions scoring for this super egg session."""
+        try:
+            message = await self.message.channel.get_message(self.message.id)
+        except discord.NotFound:
+            return
+
+        count = 0
+        white = 0
+        blurple = 0
+        react_users = []
+        for reaction in message.reactions:
+            if reaction.emoji == self.egg:
+                react_users = await reaction.users().flatten()
+                for user in react_users:
+                    team = get_team_role(user)
+                    if team == Roles.white:
+                        white += 1
+                    elif team == Roles.blurple:
+                        blurple += 1
+                count = reaction.count - 1
+                break
+
+        score = 50 if self.egg == Emoji.egg_gold else 100
+        if white == blurple:
+            log.debug("Tied SuperEgg Result.")
+            team = None
+            score = score / 2
+        elif white > blurple:
+            team = Roles.white
+        else:
+            team = Roles.blurple
+
+        embed = self.message.embeds[0]
+
+        db = sqlite3.connect(DB_PATH)
+        c = db.cursor()
+
+        user_bonus = 5 if self.egg == Emoji.egg_gold else 10
+        for user in react_users:
+            if user.bot:
+                continue
+            role = get_team_role(user)
+            if not role:
+                print('issue')
```

Comment:
Seems like double quotes are used pretty much everywhere except here from what I've seen so far. Might change it for the sake of consistency.

Reply:
Yeah, project style is double quotes everywhere unless required to avoid. Will fix that, thanks.